src/routes/ConfigureQuiz/QuestionCard.tsx

```diff
@@ -93,43 +94,73 @@ export default function QuestionCard({
     updateQuestionData(idx, question);
     await updateQuizData();
     setActiveQuestionIndex(null);
-    setExpandedQuestionIndex(idx);
+    setExpandedQuestionId(question.questionId);
 
     showAlert({
       message: 'Question has been saved successfully.',
       type: 'success',
     });
   }
 
+  function onQuestionSwap(data) {
+    const updatedQuestionIds = data.map((x) => x.questionId).join(',');
+    const initialQuestionIds = questions.map((x) => x.questionId).join(',');
+
+    if (initialQuestionIds !== updatedQuestionIds) {
+      replace(
+        data.map((x) => {
+          delete x.id;
+          return x;
+        }),
+      );
+    }
+  }
+
   return (
     <Card shadow="sm" withBorder className={`fullHeight primaryCard ${styles.questionsCard}`}>
       <Title order={5} mb="xl">
         {activeCategoryName || 'Unnamed Category'}
       </Title>
-      {questions.map((item: any, idx) =>
-        activeQuestionIndex === idx ? (
-          <QuestionEdit
-            questionNum={idx + 1}
-            question={item}
-            key={item.questionId}
-            saveQuestion={(data) => handleSaveQuestion(idx, data)}
-            onQuestionChange={(data) => updateQuestionData(idx, data)}
-            deleteQuestion={(ev) => handleDeleteQuestion(ev, idx)}
-            resetQuestion={resetQuestion}
-          />
-        ) : (
-          <QuestionView
-            questionNum={idx + 1}
-            question={item}
-            key={item.questionId}
-            isValidQuestion={isValidQuestion(item)}
-            setActiveQuestion={(ev) => setActiveQuestionIndex(idx)}
-            deleteQuestion={(ev) => handleDeleteQuestion(ev, idx)}
-            isExpanded={expandedQuestionIndex === idx}
-            setExpandedQuestionIndex={setExpandedQuestionIndex}
-          />
-        ),
-      )}
+      <ReactSortable
+        list={questions.map((item, idx) => ({ ...item, id: idx + 1, name: item.text }))}
+        chosenClass={styles.chosenStyle}
+        handle=".questionHandle"
+        onChoose={(data, ev: any) => {
+          if (!!activeQuestionIndex) {
```

Comment:
We won't need this if we don't allow dragging in expanded questions.

Reply:
I'm not allowing dragging any question. if a question is in edit mode